lib/facility.rb

```diff
@@ -1,14 +1,189 @@
+    @name = hash[:name]
+    @address = hash[:address]
+    @phone = hash[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+  #refactor
+  def register_vehicle(car)
+    if services.find {|service| service.include?('Vehicle Registration')}
+    current_time = DateTime.now
+    cdt = current_time.strftime "%d/%m/%Y %H:%M"
+    car.registration_date = cdt
+    @registered_vehicles << car
+
+      if Date.today.year - car.year > 25
+        car.plate_type = :antique
+      elsif Date.today.year - car.year <= 25 && car.engine == :ice
+        car.plate_type = :regular
+      else
+        car.plate_type = :ev
+      end
+
+      if car.plate_type == :antique
+        @collected_fees += 25
+      elsif car.engine == :ev
+        @collected_fees += 200
+      else
+        @collected_fees =+ 100
+      end
+    else
+      nil
+    end
+  end
+
+  def administer_written_test(name)
+    if services.find {|service| service.include?('Written Test')}
+      if name.permit == true && name.age >= 16
+        name.license_data[:written] = true
+        true
+      else
+        name.permit == false || name.age < 16
+        false
+      end
+    else
+      false
+    end
+  end
+
+  def administer_road_test(name)
+    if services.find {|service| service.include?('Road Test')}
+      if name.license_data[:written] == true
+        name.license_data[:license] = true
+        true
+      end
+    else
+      false
+    end
+  end
+
+  def renew_drivers_license(name)
+    if services.find {|service| service.include?('Renew License')}
+      if name.license_data[:written] == true && name.license_data[:license] == true
+        name.license_data[:renewed] = true
+      elsif name.license_data[:written] == false || name.license_data[:license] == false
+        name.license_data[:renewed] = false
+      end
+    else
+      false
+    end
+  end
+
+  def create_facility(list) #filter by something more permanent
+    nf_arr = []
+    if list.count == 5 || list.find {|omv| omv[:state] == "CO"} #CO
+      list.each do |omv|
+        new_omv = {
+          name: omv[:dmv_office],
+          address: omv[:address_li], #+ ", " + omv[:address__1],
+          phone: omv[:phone]
+        }
+        nf_arr << Facility.new(new_omv)
+      end
+    elsif list.count == 179 || list.find {|omv| omv[:state] == "MO"} #MO
+      list.each do |omv|
+        new_omv = {
+          name: omv[:name],
+          address: omv[:address1],
+          phone: omv[:phone]
+        }
+        nf_arr << Facility.new(new_omv)
+      end
+    elsif list.count == 172 || list.find {|omv| omv[:state] == "NY"} #NY
+      list.each do |omv|
+        new_omv = {
+          name: omv[:office_name],
+          address: omv[:street_address_line_1],
+          phone: omv[:public_phone_number]
+        }
+        nf_arr << Facility.new(new_omv)
+      end
+    else
```

Comment:
I don't love how you're checking the count of the list. What if NY suddenly opened 7 more DMV's? Then they'd get dropped into the block on line 110-118 because it now has 179 DMV offices. I do, however, like your idea of checking the state. That seems like a better approach for formatting rather than the count of the offices.

Reply:
Ahhh I didn't think about what would happen if they had the same count!